cub/cub/util_device.cuh

```diff
@@ -28,7 +28,11 @@
 
 /**
  * \file
- * Properties of a given CUDA device and the corresponding PTX bundle
+ * Properties of a given CUDA device and the corresponding PTX bundle.
+ *
+ * \note
+ * This file contains __host__ only functions and utilities, and should not be
+ * included in code paths that could be online-compiled (ex: using NVRTC).
```

Comment:
important: it's unfortunate that CUB has a fuzzy API border. The `AliasTemporaries` in a public namespace and used to be provided by this header. Just to be on the safe side and not break users who included this header specifically for the `AliasTemporaries`, do you think we could include `util_temporary_storage.cuh` here? Also, a comment describing that the include is needed for the compatibility reasons is welcome.

Reply:
Yeah, for some reason I left a comment asking for exactly the same question, but forgot to hit send 🙂

I have restored it in commit 91c4556c to restore backward compatibility. That said, `AliasTemporaries` is supposed to be a private API (it's undocumented as it's sandwiched in `DOXYGEN_SHOULD_SKIP_THIS`), so I am not sure it's breaking anyone.